### PR TITLE
[Merged by Bors] - feat: `divRadical` for polynomials

### PR DESCRIPTION
**Pull request by seewoo5**
---

Some theorems on `divRadical` for polynomials.

- [x] depends on: #18449

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Polynomial/Radical.lean [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+noncomputable section
+
+open scoped Classical
+
+open Polynomial UniqueFactorizationMonoid UniqueFactorizationDomain EuclideanDomain
+
+variable {k : Type*} [Field k]
+
+
+theorem divRadical_dvd_derivative (a : k[X]) : (divRadical a) ∣ (derivative a) := by
```
Comment (YaelDillies) [inline Mathlib/RingTheory/Polynomial/Radical.lean:26]:
```suggestion
theorem divRadical_dvd_derivative (a : k[X]) : divRadical a ∣ derivative a := by
```

File: Mathlib/RingTheory/Polynomial/Radical.lean [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  · -- If it holds for coprime pair a and b, then it also holds for a * b.
+    case hcp x y hpxy hx hy =>
+    have hc : IsCoprime x y :=
+      EuclideanDomain.isCoprime_of_dvd
+        (fun ⟨hx, hy⟩ => not_isUnit_zero (hpxy (zero_dvd_iff.mpr hx) (zero_dvd_iff.mpr hy)))
+        fun p hp _ hpx hpy => hp (hpxy hpx hpy)
+    rw [divRadical_mul hc, derivative_mul]
+    exact dvd_add (mul_dvd_mul hx (divRadical_dvd_self y)) (mul_dvd_mul (divRadical_dvd_self x) hy)
+
+theorem divRadical_dvd_wronskian_left (a b : k[X]) : (divRadical a) ∣ wronskian a b := by
```
Comment (YaelDillies) [inline Mathlib/RingTheory/Polynomial/Radical.lean:53]: Same here and below

File: Mathlib/RingTheory/Polynomial/Radical.lean [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+noncomputable section
+
+open scoped Classical
+
+open Polynomial UniqueFactorizationMonoid UniqueFactorizationDomain EuclideanDomain
+
+variable {k : Type*} [Field k]
+
+
```
Comment (YaelDillies) [inline Mathlib/RingTheory/Polynomial/Radical.lean:25]:
```suggestion
```

**Review by YaelDillies — COMMENTED** [review]
maintainer merge

apart from that. Thanks!

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**